packages/api/client.ts

```diff
@@ -68,22 +68,49 @@ export const getOperators = async (dbClient: Kysely<Database>, input: OperatorQu
 export type StopsQueryInput = {
     atcoCodes?: string[];
     naptanCodes?: string[];
+    commonName?: string;
+    adminAreaCode?: string;
     page?: number;
 };
 
 export const getStops = async (dbClient: Kysely<Database>, input: StopsQueryInput) => {
     logger.info("Starting getStops...");
 
-    const STOPS_PAGE_SIZE = 50;
+    const STOPS_PAGE_SIZE = 1000;
 
-    if (input.atcoCodes || input.naptanCodes) {
+    if (input.adminAreaCode) {
```

Comment:
I'm wondering if we could actually do everything in one with something like:
```ts
export const getStops = async (dbClient: Kysely<Database>, input: StopsQueryInput) => {
    logger.info("Starting getStops...");
    logger.info(process.env.NODE_ENV ?? "");

    const STOPS_PAGE_SIZE = 50;

    logger.info("Filtering by admin area code...");
    const stops = await dbClient
        .selectFrom("stops")
        .selectAll()
        .$if(!!input.atcoCodes?.[0], (qb) => qb.where("atcoCode", "in", input.atcoCodes ?? ["---"]))
        .$if(!!input.naptanCodes?.[0], (qb) => qb.where("naptanCode", "in", input.naptanCodes ?? ["---"]))
        .$if(!!input.commonName?.[0], (qb) =>
            qb.where("commonName", "like", input.commonName ? `%${input.commonName}%` : "---"),
        )
        .$if(!!input.adminAreaCodes?.[0], (qb) =>
            qb.where("administrativeAreaCode", "in", input.adminAreaCodes ?? ["---"]),
        )
        .offset((input.page || 0) * STOPS_PAGE_SIZE)
        .limit(STOPS_PAGE_SIZE)
        .execute();

    if (!stops) {
        return null;
    }

    return stops;
};
```
assuming we make adminAreaCodes a query param array. This will also make the conditions ands rather than ors which on reflection probably makes more sense when it comes to filtering down data.

Reply:
done